src/controllers/UserController.ts

```diff
@@ -0,0 +1,86 @@
+import {Request, Response, NextFunction} from "express"
+import User from "../models/users"
+
+import {
+	UpdateUserPayload,
+	ListUserPayload,
+	DeleteUserPayload
+} from "../types/auth"
+import {UserDetails, UserUpdatePayload} from "../types/users"
+import helper, {generateOtp, sendSMS, decryptBycrypto} from "../helpers/helper"
+import {ApiResponse} from "../helpers/ApiResponse"
+import { BadRequestException } from "../lib/exceptions"
+// import errorData from "../constants/errorData.json"
+
+class AuthController {
+	constructor() {
+		
+		this.list = this.list.bind(this)
+		this.update = this.update.bind(this)
+		this.delete = this.delete.bind(this)
+	}
+
+	public async update(req: Request, res: Response, next: NextFunction) {
+		try {
+			const response = new ApiResponse(res)
+			const inputData: UserUpdatePayload = req.body
+
+			const listUserData = await User.findOne({userId: inputData.userId})
+			if (!listUserData) {
+				throw new Error("User not found")
+			}
+
+		// update  
+		const data = await User.updateOne({userId: inputData.userId} , {$set: inputData})
+		return response.successResponse({
+			message : "User updated successfully",
```

Comment:
Add one extra validation.
if email to update set isVerified to false.

Reply:
can't we add it in update isVerified as well?
like

const data = await User.findByIdAndUpdate({userId: inputData.userId} , {$set: inputData, $isVerified: false})

Reply:
Just add isVerified to inputData object